obsplus/events/validate.py

```diff
@@ -100,57 +99,191 @@ def check_origins(event: Event):
+        if len(p_picks) and len(amp_picks):
+            ptime = p_picks.iloc[0]["time"]
+            bad_amp_picks = amp_picks[amp_picks["time"] < ptime]
+            ap.extend(list(bad_amp_picks["seed_id"]))
+
+    # Checking that picks are in acceptable order
+    gb, sp, ap = pdf.groupby("station"), [], []
+    gb.apply(pick_order, sp, ap)
+    assert len(sp) == 0, "S pick found before P pick:\n" f"station/s: {sp}"
+    assert len(ap) == 0, "amplitude pick found before P pick:\n" f"seed_id/s: {ap}"
+
+
+@validator("obsplus", Event)
+def check_p_lims(event: Event, p_lim=None, **kwargs):
+    """
+    Check for P picks that aren't within p_lim of the median pick (if provided)
+    """
+    if p_lim is not None:
+        df = obsplus.picks_to_df(event)
+        df = df.loc[(df.evaluation_status != "rejected") & (df.phase_hint == "P")]
+        med = df.time.median()
+        bad = df.loc[abs(df.time - med) > p_lim]
+        assert len(bad) == 0, (
+            "Outlying P pick found:\n"
+            f"event_id: {event.resource_id.id}, "
+            f"seed_id/s: {bad.seed_id.tolist()}"
+        )
+
+
+@validator("obsplus", Event)
+def check_amp_lims(event: Event, amp_lim=None, **kwargs):
+    """
+    Check for amplitudes that aren't below amp_lim (if provided)
+    """
+    if amp_lim is not None:
+        bad = []
+        for amp in event.amplitudes:
+            if amp.generic_amplitude > amp_lim:
+                wid = amp.waveform_id
+                nslc = (
+                    f"{wid.network_code}.{wid.station_code}."
+                    f"{wid.location_code}.{wid.channel_code}"
+                )
+                bad.append(nslc)
+        assert len(bad) == 0, (
+            "Above limit amplitude found:\n"
+            f"event_id: {event.resource_id.id}, "
+            f"seed_id/s: {bad}"
+        )
+
+
+@validator("obsplus", Event)
+def check_amp_filts(event: Event, filt_amps=None, **kwargs):
+    """
+    Check that all amplitudes have a specified filter id (if provided)
+    """
+    if filt_amps is not None:
+        if type(filt_amps) is ResourceIdentifier:
+            filt_amps = filt_amps.id
+        bad = []
+        bad_filters = []
+        for amp in event.amplitudes:
+            if amp.filter_id.id != filt_amps:
+                wid = amp.waveform_id
+                nslc = (
+                    f"{wid.network_code}.{wid.station_code}."
+                    f"{wid.location_code}.{wid.channel_code}"
+                )
+                bad.append(nslc)
+                if amp.filter_id.id not in bad_filters:
+                    bad_filters.append(amp.filter_id.id)
+        assert len(bad) == 0, (
+            "Unexpected amplitude filter found:\n"
+            f"event_id: {event.resource_id.id}, "
+            f"seed_id/s: {bad}, "
+            f"filters_used: {set(bad_filters)}"
+        )
+
+
+@validator("obsplus", Event)
+def check_z_amps(event: Event, no_z_amps=False, **kwargs):
```

Comment:
I once again think this could benefit from being able to specify which amplitude types/magnitude hints to apply this to in order to allow it to work on more than just "AML" amplitudes.